x/dist/keeper/allocation.go

```diff
@@ -29,27 +29,27 @@ func (k Keeper) AllocateTokens(ctx sdk.Context, blockProposer sdk.ConsAddress) {
 
 	/* ---------------------------- Pay the proposer ---------------------------- */
 	// calculate and pay proposer reward
-	proposer, found := k.seqKeeper.GetSequencerByConsAddr(ctx, blockProposer)
+	addr, found := k.seqKeeper.GetRewardAddrByConsAddr(ctx, blockProposer)
 	if !found {
-		logger.Error("failed to find the validator for this block. reward not allocated")
+		logger.Error("Find the validator for this block. Reward not allocated.", "addr", blockProposer)
 	} else {
 		proposerReward := feesCollected.MulDecTruncate(k.GetBaseProposerReward(ctx))
 		proposerCoins, proposerRemainder := proposerReward.TruncateDecimal()
-
-		err := k.AllocateTokensToSequencer(ctx, proposer, proposerCoins)
-		if err != nil {
-			logger.Error("failed to reward proposer", "error", err, "proposer", proposer.GetOperator())
-		} else {
-			remainingFees = feesCollected.Sub(proposerReward).Add(proposerRemainder...)
-
-			// update outstanding rewards
-			ctx.EventManager().EmitEvent(
-				sdk.NewEvent(
-					disttypes.EventTypeDistSequencerRewards,
-					sdk.NewAttribute(sdk.AttributeKeyAmount, proposerCoins.String()),
-					sdk.NewAttribute(disttypes.AttributeKeySequencer, proposer.GetOperator().String()),
-				),
-			)
+		if !proposerCoins.IsZero() {
```

Comment:
better to early exit `if proposerCoins.IsZero()`

Reply:
Why? Then staking validators wont get anything